src/openrct2-ui/interface/Dropdown.h

```diff
@@ -46,27 +35,45 @@ namespace OpenRCT2::Ui::Windows
     extern int32_t gDropdownDefaultIndex;
 
     void WindowDropdownShowText(
-        const ScreenCoordsXY& screenPos, int32_t extray, ColourWithFlags colour, uint8_t flags, size_t num_items);
+        const ScreenCoordsXY& screenPos, int32_t extray, ColourWithFlags colour, uint8_t flags, size_t num_items,
+        size_t prefRowsPerColumn = 0);
     void WindowDropdownShowTextCustomWidth(
         const ScreenCoordsXY& screenPos, int32_t extray, ColourWithFlags colour, uint8_t custom_height, uint8_t flags,
-        size_t num_items, int32_t width);
+        size_t num_items, int32_t width, size_t prefRowsPerColumn = 0);
+
     void WindowDropdownShowImage(
         int32_t x, int32_t y, int32_t extray, ColourWithFlags colour, uint8_t flags, int32_t numItems, int32_t itemWidth,
         int32_t itemHeight, int32_t numColumns);
+
     void WindowDropdownClose();
+
     int32_t DropdownIndexFromPoint(const ScreenCoordsXY& loc, WindowBase* w);
+
     void WindowDropdownShowColour(
         WindowBase* w, Widget* widget, ColourWithFlags dropdownColour, colour_t selectedColour,
         bool alwaysHideSpecialColours = false);
     void WindowDropdownShowColourAvailable(
         WindowBase* w, Widget* widget, uint8_t dropdownColour, uint8_t selectedColour, uint32_t availableColours);
-    uint32_t DropdownGetAppropriateImageDropdownItemsPerRow(uint32_t numItems);
 
     colour_t ColourDropDownIndexToColour(uint8_t ddidx);
+
+    uint32_t DropdownGetAppropriateImageDropdownItemsPerRow(uint32_t numItems);
 } // namespace OpenRCT2::Ui::Windows
 
 namespace OpenRCT2::Dropdown
 {
+    bool IsChecked(int32_t index);
+    bool IsDisabled(int32_t index);
+    void SetChecked(int32_t index, bool value);
+    void SetDisabled(int32_t index, bool value);
+    void SetImage(int32_t index, ImageId image);
+
+    enum Flag
+    {
+        CustomHeight = (1 << 6), // never set?
```

Comment:
Most likely, this was used to force multiple columns for the Ride type and vehicle type dropdowns, back when the dropdowns didn’t automatically use multiple columns if there wasn’t enough space.

Reply:
Interesting. I say we leave it for now and revisit it when we do a bigger (state) refactor of the dropdowns.